flink-runtime/src/main/java/org/apache/flink/runtime/metrics/dump/MetricDumpSerialization.java

```diff
@@ -50,12 +51,27 @@
 	private MetricDumpSerialization() {
 	}
 
+	public static class MetricSerializationResult {
+		public final byte[] data;
+		public final int numCounters;
+		public final int numGauges;
+		public final int numMeters;
+		public final int numHistograms;
+		
+		public MetricSerializationResult(byte[] data, int numCounters, int numGauges, int numMeters, int numHistograms) {
+			this.data = data;
```

Comment:
Let's add simple sanity checks `checkNotNull(data)` and `checkArgument(num* >= 0)` for the other fields.